src/main/scala/is/hail/io/annotators/BedAnnotator.scala

```diff
@@ -54,22 +55,27 @@ object BedAnnotator {
           line.value.startsWith("track") ||
           line.value.matches("""^\w+=("[\w\d ]+"|\d+).*""") ||
           line.value.isEmpty))
-      .map {
+      .flatMap {
         _.map { line =>
           val spl = line.split("""\s+""")
           if (spl.length < expectedLength)
             fatal(s"Expected at least $expectedLength fields, but found ${ spl.length }")
           val chrom = spl(0)
           // transform BED 0-based coordinates to Hail/VCF 1-based coordinates
           val start = spl(1).toInt + 1
-          val end = spl(2).toInt + 1
+          val end = spl(2).toInt
 
-          val interval = Interval(Locus.annotation(chrom, start, rg), Locus.annotation(chrom, end, rg), true, false)
+          if (skipInvalidIntervals && rg.forall(rg => !rg.isValidContig(chrom) || !rg.isValidLocus(chrom, start) || !rg.isValidLocus(chrom, end))) {
+            warn(s"Interval is not consistent with reference genome '${ rg.get.name }': '$line'.")
```

Comment:
this won't work if it's happening on an executor, I think.

Reply:
Why not?

Reply:
the warning will go to executor logs, it won't propagate back to the master.

Reply:
can we delete this warning?